test/fitnesse/junit/JUnitXMLPerPageRunListenerIntegrationTest.java

```diff
@@ -0,0 +1,103 @@
+package fitnesse.junit;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.Description;
+import org.junit.runner.notification.Failure;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class JUnitXMLPerPageRunListenerIntegrationTest {
+
+  @Rule
+  public TemporaryFolder reportDir = new TemporaryFolder();
+
+  private JUnitXMLReportHelper jUnitXMLReportHelper;
+
+  // SUT
+  private JUnitXMLPerPageRunListener jUnitXMLPerPageRunListener;
+
+  @Before
+  public void setUp() {
+    jUnitXMLPerPageRunListener = new JUnitXMLPerPageRunListener(new JUnitXMLTestResultRecorder(reportDir.getRoot()));
+    jUnitXMLReportHelper = new JUnitXMLReportHelper(reportDir.getRoot());
+  }
+
+  @Test
+  public void testSuccess() {
+    // given some test
+    Description description = Description.createTestDescription(JUnitXMLPerPageRunListenerIntegrationTest.class, "myTestName");
+
+    // when the test is started and finished
+    try {
+      jUnitXMLPerPageRunListener.testStarted(description);
+      jUnitXMLPerPageRunListener.testFinished(description);
+    } catch (Exception e) {
```

Comment:
Instead of catching an exception only to fail the test you can just add a `throws` clause to the method definitions...

Reply:
Obsolete. Deleted integration test, kept unit test.